### PR TITLE
Create note listing page UI

### PR DESCRIPTION
**Pull request by rishiip**
SCR-20220818-pgv

### CODE REVIEW

File: app/javascript/src/components/Dashboard/Notes/Note.jsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import React from "react";
+
+import dayjs from "dayjs";
+import { MenuVertical, Clock } from "neetoicons";
+import { Typography, Dropdown, Tag, Tooltip, Avatar } from "neetoui";
+
+var relativeTime = require("dayjs/plugin/relativeTime");
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/Note.jsx:7]: Why's this declared as a `var`?

File: app/javascript/src/components/Dashboard/Notes/Note.jsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+dayjs.extend(relativeTime);
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/Note.jsx:9]: I am not quite sure why we are extending this plugin in this component. Is it because only this particular component needs to make use of the methods from `relativeTime`? Isn't the `dayjs` instance globally available? If yes, then this is not the right place add the initialization.

File: app/javascript/src/components/Dashboard/Notes/NoteMenuBar.jsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  const [isSearchCollapsed, setIsSearchCollapsed] = useState(true);
+
+  return (
+    <div className="flex">
+      <MenuBar showMenu={showMenu} title="Notes">
+        <MenuBar.Block label="All" count={13} active />
+        <MenuBar.Block label="Users" count={2} />
+        <MenuBar.Block label="Leads" count={7} />
+        <MenuBar.Block label="Visitors" count={4} />
+
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/NoteMenuBar.jsx:17]: It seems to me like you haven't added the latest eslint configs from wheel. Go through the steps mentioned in https://www.bigbinary.com/books/learn-rubyonrails-book/linting-and-formatting-code and pull in the latest rules and apply those to all files in this PR.

Stressing on the part `files in this PR`. I don't want you to apply those rules on all files in your repo which are not part of this PR.

File: app/javascript/src/components/Dashboard/Notes/NoteMenuBar.jsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        <MenuBar.SubTitle
+          iconProps={[
+            {
+              icon: Search,
+              onClick: () => setIsSearchCollapsed(!isSearchCollapsed),
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/NoteMenuBar.jsx:22]: https://academy.bigbinary.com/learn-react/pages/know-when-to-use-previous-state-for-state-update

**Review by yedhink — COMMENTED** [review]
@rishiip I've added some comments.

- I have not pinpointed each and every spot a similar change ought to be made. Thus when you work on a review comment, please make sure to check your whole PR's files to see where a similar change needs to be made.
- You can make these changes as part of your next feature PR or raise a separate PR just fixing all the review changes.

**Comment by rishiip** [comment]
@yedhink - Raised PR for above review suggestions - #12

File: app/javascript/src/components/Dashboard/Notes/Note.jsx (hunk 2 above)
Comment (rishiip) [reply app/javascript/src/components/Dashboard/Notes/Note.jsx:9]: Yes, we're using methods from relativeTime plugin over here only. `daysjs` instance is not globally available here so extending it over here only.

File: app/javascript/src/components/Dashboard/Notes/NoteMenuBar.jsx (hunk 3 above)
Comment (rishiip) [reply app/javascript/src/components/Dashboard/Notes/NoteMenuBar.jsx:17]: This is done.

File: app/javascript/src/components/Dashboard/Notes/NoteMenuBar.jsx (hunk 4 above)
Comment (rishiip) [reply app/javascript/src/components/Dashboard/Notes/NoteMenuBar.jsx:22]: Thanks for pointing this out. This is done.

File: app/javascript/src/components/Dashboard/Notes/index.jsx [hunk 5]
```diff
@@ -1,102 +1,51 @@
...
-import { Button, PageLoader } from "neetoui";
-import { Container, Header, SubHeader } from "neetoui/layouts";
+import { Button } from "neetoui";
+import { Container, Header } from "neetoui/layouts";
 
-import notesApi from "apis/notes";
-import EmptyState from "components/Common/EmptyState";
-
-import DeleteAlert from "./DeleteAlert";
+import { NOTES as notes } from "./constants";
```
[the remark this answers is not in the document]
Comment (rishiip) [reply app/javascript/src/components/Dashboard/Notes/index.jsx:6]: It's fixed. Keeping `NOTES` only so that we can identify that it's constant.

File: app/javascript/src/components/Dashboard/Notes/index.jsx [hunk 6]
```diff
@@ -1,102 +1,51 @@
...
-      setLoading(true);
-      const { data } = await notesApi.fetch();
-      setNotes(data.notes);
-    } catch (error) {
-      logger.error(error);
-    } finally {
-      setLoading(false);
-    }
+  const fetchNotes = () => {
+    [];
```
[the remark this answers is not in the document]
Comment (rishiip) [reply app/javascript/src/components/Dashboard/Notes/index.jsx:17]: This is removed as we're using static NOTES. So this is not needed.

File: app/javascript/src/components/Dashboard/Notes/Note.jsx (hunk 2 above)
Comment (yedhink) [reply app/javascript/src/components/Dashboard/Notes/Note.jsx:9]: >daysjs instance is not globally available

I am not sure based on what factors or experiments you are asserting this. Can you please put up a video showing me how you made the judgement that the `dayjs` instance is NOT global?  Stressing on the `video` part.  I would recommend reading and understanding what `extend` does underneath the hood.

Comment (yedhink) [reply app/javascript/src/components/Dashboard/Notes/Note.jsx:9]: @rishiip You don't have to create another instance for the extended version. 

>I would recommend reading and understanding what extend does underneath the hood.

The `extend` method would extend into the global instance of `dayjs`. So create a `src/lib/dayjs.js` and extend all the necessary dayjs plugins over there. Then import it once in say `Main.jsx`.

Currently you've kept the `extend` part with a constants file scoped under `Notes` namespace. Which necessarily doesn't constraint the extension to be within that namespace because the `dayjs` instance is global.

Comment (rishiip) [reply app/javascript/src/components/Dashboard/Notes/Note.jsx:9]: @yedhink - Thanks for the suggestions. I added the required modifications and pushed in [this commit](https://github.com/rishiip/wheel/pull/12/commits/35477b42bf6237d278bd1801a30486e95b97415d). Have  a look.
